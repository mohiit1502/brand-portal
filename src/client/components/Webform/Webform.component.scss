.c-Webform {
    .header{
        font-weight: bold;
        padding-bottom: 1.5rem;
    }

    .btn {
        width: 150px;
    }

}
