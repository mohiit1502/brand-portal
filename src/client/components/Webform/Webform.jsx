/* eslint-disable max-statements */
/* eslint-disable filenames/match-regex */
import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import ContentRenderer from "../../utility/ContentRenderer";
import "./WebForm.component.scss";
import Helper from "../../utility/helper";
import Validator from "../../utility/validationUtil";
import CONSTANTS from "../../constants/constants";
import InputFormatter from "../../utility/phoneOps";


class WebForm extends React.Component {
  constructor(props) {
    super(props);
    const functions = ["onChange", "setSelectInputValue", "undertakingtoggle", "getClaimTypes", "selectHandlersLocal", "checkToEnableSubmit", "customChangeHandler", "getItemListFromChild", "bubbleValue", "handleSubmit"];
    functions.forEach(name => this[name] = this[name].bind(this));

    const debounceFunctions = {emailDebounce: "onEmailChange"};
    Object.keys(debounceFunctions).forEach(name => {
      const functionToDebounce = Validator[debounceFunctions[name]] ? Validator[debounceFunctions[name]].bind(this) : this[debounceFunctions[name]];
      this[name] = Helper.debounce(functionToDebounce, CONSTANTS.APIDEBOUNCETIMEOUT);
    });
    this.onInvalid = Validator.onInvalid.bind(this);
    this.invalid = {emailId: false, phone: false};
    const webformConfiguration = this.props.webformConfiguration ? this.props.webformConfiguration : {};
    this.getFieldRenders = ContentRenderer.getFieldRenders.bind(this);
    this.itemUrlDebounce = Helper.debounce(this.onItemUrlChange, CONSTANTS.APIDEBOUNCETIMEOUT);
    this.state = {
      section: {...webformConfiguration.sectionConfig},
      form: {
        ...webformConfiguration.formConfig,
        inputData: {...webformConfiguration.fields}
      }
    };
    const formatter = new InputFormatter();
    const handlers = formatter.on(`#${this.state.section.id}-${this.state.form.inputData.phone.inputId}-custom-input`);
    this.prebounceChangeHandler = handlers.inputHandler;
  }

  componentDidMount() {
    this.getClaimTypes();
  }

  getClaimTypes () {
    const state = {...this.state};
    const form = state.form;
    const options = [
      {claimType: "trademark", label: "Trademark", claimTypeIdentifierLabel: "Trademark Number", companyNameIdentifierLabel: "Trademark Company Name", ownerNameIdentifierLabel: "Trademark Owner Name"},
      {claimType: "patent", label: "Patent", claimTypeIdentifierLabel: "Patent Number", companyNameIdentifierLabel: "Patent Company Name", ownerNameIdentifierLabel: "Patent Owner Name"},
      {claimType: "counterfeit", label: "Counterfeit", claimTypeIdentifierLabel: "Trademark Number", companyNameIdentifierLabel: "Rights Owner Company Name", ownerNameIdentifierLabel: "Rights Owner Name"},
      {claimType: "copyright", label: "Copyright", claimTypeIdentifierLabel: "Copyright Number", companyNameIdentifierLabel: "Copyright Company Name", ownerNameIdentifierLabel: "Copyright Owner Name"}
    ];
        form.inputData.claimType.claimTypesWithMeta = options;
        form.inputData.claimType.dropdownOptions = options && options.map(v => ({value: v.label}));
        this.setState(state);
  }

  getItemListFromChild(itemList) {
    this.setState(state => {
      state = {...state};
      state.form.inputData.urlItems.itemList = [...itemList];
      return {
        ...state
      };
    });
  }

  bubbleValue (evt, key, error) {
    const targetVal = evt.target.value;
    this.setState(state => {
      state = {...state};
      state.form.inputData[key].value = targetVal;
      state.form.inputData[key].error = error;
      return state;
    }, this.checkToEnableSubmit);
  }

  onChange (evt, key) {
    if (evt && evt.target) {
      evt.target.checkValidity();
      const targetVal = evt.target.value;
      let index = -1;
      if (key.split("-")[0] === "url" || key.split("-")[0] === "sellerName" && key.split("-")[1]) {
        index = Number(key.split("-")[1]);
        key = key.split("-")[0];
      }

      this.setState(state => {
        state = {...state};
        if (index > -1) {
          if (key.split("-")[0] === "url") {
            state.form.inputData.urlItems.itemList[index].sellerName.value = "";
            state.form.inputData.urlItems.itemList[index].sellerName.disabled = false;
            state.form.inputData.urlItems.itemList[index][key].value = targetVal;
            state.form.inputData.urlItems.disableAddItem = true;
          } else  {
            state.form.inputData.urlItems.itemList[index][key].value = targetVal;
            state.form.inputData.urlItems.itemList[index][key].error = "";
            state.form.inputData.urlItems.itemList[index].url.error = "";
            state.form.inputData.urlItems.disableAddItem = false;
          }
        } else {
          state.form.inputData[key].error = !this.invalid[key] ? "" : state.form.inputData[key].error;
          state.form.inputData[key].value = targetVal;
          this.invalid[key] = false;
        }
        return {
          ...state
        };
      }, this.checkToEnableSubmit);
      evt.persist();
      if (key === "url") {
        this.itemUrlDebounce(evt, index);
      }
    }
  }
  onItemUrlChange (event, i) {
    let url = event && event.target.value;
    if (url) {
      this.loader("fieldLoader", true);
      if (url.endsWith("/")) {
        url = url.substring(0, url.length - 1);
      }
    }
  }
  selectHandlersLocal (key, state, value) {
    return state;
  }

  customChangeHandler (value) {
    const form = this.state.form;
    const claimTypesWithMeta = form.inputData.claimType.claimTypesWithMeta;
    const matchedClaimTypeWithMeta = claimTypesWithMeta.find(claimTypeWithMeta => claimTypeWithMeta.label === value);
    if (matchedClaimTypeWithMeta) {
      form.inputData.companyName.label = matchedClaimTypeWithMeta.companyNameIdentifierLabel;
      form.inputData.ownerName.label = matchedClaimTypeWithMeta.ownerNameIdentifierLabel;
      form.claimTypeSelected = true;
      this.setState({form});
    }
  }

  setSelectInputValue (value, key) {
    if (value) {
      let index = -1;
      if (key.split("-")[0] === "sellerName" && key.split("-")[1]) {
        index = Number(key.split("-")[1]);
        key = key.split("-")[0];
      }
      this.setState(state => {
        state = {...state};
        state = this.selectHandlersLocal(key, state, value);
        state.form.inputData[key].value = value;
        return {
          ...state
        };
      }, () => this.checkToEnableSubmit());
    }
  }

  // eslint-disable-next-line complexity
  checkToEnableSubmit(callback) {
    const form = {...this.state.form};
    const userUndetaking = form.inputData.user_undertaking_1.selected && form.inputData.user_undertaking_2.selected && form.inputData.user_undertaking_3.selected && form.inputData.user_undertaking_4.selected;
    const isValidItemList = form.inputData.urlItems.itemList.reduce((boolResult, item) => !!(boolResult && item.url.value && !item.url.error && item.sellerName.value && item.sellerName.value.length > 0 && !item.sellerName.error), true);
    const bool = isValidItemList && userUndetaking  && form.inputData.claimType.value &&
      form.inputData.firstName.value && form.inputData.lastName.value &&
      form.inputData.ownerName.value && form.inputData.companyName.value &&
      form.inputData.brandName.value &&
      form.inputData.address_1.value && form.inputData.address_2.value &&
      form.inputData.city.value && form.inputData.country.value &&
      form.inputData.state.value && form.inputData.zip.value && !form.inputData.zip.error &&
      form.inputData.phone.value && !form.inputData.phone.error &&  form.inputData.emailId.value && !form.inputData.emailId.error &&
      form.inputData.comments.value && !form.inputData.comments.error &&
      form.inputData.digitalSignature.value;

    form.isSubmitDisabled = !bool;
    form.inputData.webFormActions.buttons.submit.disabled = !bool;
    this.setState({form}, callback && callback());
  }

  undertakingtoggle (evt, undertaking, index) {
    const state = {...this.state};
    state.form.inputData[evt.target.id].selected = !state.form.inputData[evt.target.id].selected;
    this.setState({
      ...state
    }, this.checkToEnableSubmit);
  }

  handleSubmit() {

  }

  render() {
    return (
      <div className="c-WebForm">
        <div className="row h3 header pl-5">
          Walmart IP Services
        </div>
        <div className="row justify-content-center">
        <div className="col-lg-7 col-md-6 col-6 pl-3 pr-3">
          <div className="row title-row mb-4 pl-2">
              <div className="web-form-title">
                {this.state.section.sectionTitle}
              </div>
          </div>
          <form onSubmit={this.handleSubmit} className="web-form mb-4 ml-3 mr-3" >
            { this.getFieldRenders()}
          </form>
        </div>
      </div>
      </div>
    );
  }

}

WebForm.propTypes = {

};

const mapStateToProps = state => {
  return {
    webformConfiguration: state.content && state.content.metadata && state.content.metadata.SECTIONSCONFIG && state.content.metadata.SECTIONSCONFIG.WEBFORM
  };
};


export default connect(mapStateToProps)(WebForm);
