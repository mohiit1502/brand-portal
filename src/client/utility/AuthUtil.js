import authConfig from "./../config/authorizations";
import Helper from "./helper";

export default class AuthUtil {
  static isActionAccessible (action, userProfile, actedOnRole) {
    if (action && action.indexOf(".") > -1) {
      const actionConfig = Helper.search(action, authConfig);
      if (actionConfig && userProfile && userProfile.role && userProfile.role.name && typeof actionConfig === "object") {
        const role = userProfile.role.name;
        if (actionConfig.length !== undefined) {
          return actionConfig.includes(role);
        } else {
          if (Object.keys(actionConfig).includes(role) && actedOnRole && actionConfig[role].includes(actedOnRole)) {
            return true;
          }
        }
      }
    }
    return false;
  }
}
