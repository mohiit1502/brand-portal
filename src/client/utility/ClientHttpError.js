export default class ClientHttpError {

  status;
  error;
  message;

  constructor (status, error, message) {
    this.status = status;
    this.error = error;
    this.message = message;
  }

}
