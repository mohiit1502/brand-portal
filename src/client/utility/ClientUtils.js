import queryString from "query-string";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }
}
