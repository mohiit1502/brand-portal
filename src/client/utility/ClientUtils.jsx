import React from "react";
import queryString from "query-string";
import {Redirect} from "react-router";

export default class ClientUtils {

  static getQueryParams (location) {
    return queryString.parse(location.search);
  }

  static  getAllValuesFromRecursiveTree(tree) {

    if (typeof tree !== "object" || tree === null) {
      return [tree];
    }

    const list = [];

    for (const i in tree) {
      if (tree.hasOwnProperty(i)) {
        list.push(...this.getAllValuesFromRecursiveTree(tree[i]));
      }
    }
    return list;
  }

  static where (arrayList, filterObj) {
    let index = -1;
    for (let i = 0; i < arrayList.length; i++) {
      for (const j in filterObj) {
        if (filterObj.hasOwnProperty(j) && arrayList[i].hasOwnProperty(j) && filterObj[j] === arrayList[i][j]) {
          index = i;
        } else {
          index = -1;
        }
      }
      if (index !== -1) {
        return index;
      }
    }
    return index;
  }

}
