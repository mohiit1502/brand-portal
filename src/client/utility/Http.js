/* eslint-disable complexity */
/* eslint-disable no-unused-expressions */
/* eslint-disable max-params */
/* eslint-disable max-statements */
import fetch from "node-fetch";
import queryString from "query-string";
import ClientHttpError from "./ClientHttpError";
import { NOTIFICATION_TYPE } from "../actions/notification/notification-actions";
import CONSTANTS from "../constants/constants";
export default class Http {

  static async get(url, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});

    const options = {
      method: "GET"
    };
    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;

    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message, err.code);
  }

  static async post(url, data, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "POST",
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json"
      }
    };
    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;
    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message);

  }

  static async postAsFormData(url, data, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "POST",
      body: data,
      headers: { }
    };
    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;
    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message);

  }

  static async put(url, data, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "PUT",
      body: JSON.stringify(data),
      headers: {
        "Content-Type": "application/json"
      }
    };

    const response = await fetch(urlString, options);
    const {ok, status, headers} = response;
    if (ok) {
      if (headers.get("content-type").indexOf("application/json") !== -1) {
        const body = await response.json();
        callback && typeof callback === "function" && callback();
        return {status, body};
      }
      const body = await response.text();
      callback && typeof callback === "function" && callback();
      return {status, body};
    }

    this.displayToast(status, toastcallback);
    const err = await response.json();
    callback && typeof callback === "function" && callback();
    console.log(err);
    throw new ClientHttpError(status, err.error, err.message);
  }

  static async delete(url, queryParams, callback, toastcallback) {
    const urlString = queryString.stringifyUrl({url, query: queryParams});
    const options = {
      method: "DELETE",
      noInject: true
    };
    return fetch(urlString, options);
  }

  displayToast(status, toastcallback) {
    if (new RegExp(CONSTANTS.CODES.ERRORCODES.SERVERERROR).test(status)) {
      toastcallback && typeof toastcallback === "function" && toastcallback(NOTIFICATION_TYPE.ERROR, `Unable to reach our services currently, please try again in sometime.`);
    } else if (new RegExp(CONSTANTS.CODES.ERRORCODES.FOURNOTFOUR).test(status)) {
      toastcallback && typeof toastcallback === "function" && toastcallback(NOTIFICATION_TYPE.ERROR, `Requested Resource could not found.`);
    }
  }
}
