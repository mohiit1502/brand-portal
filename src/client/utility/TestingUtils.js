import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../reducers";
import thunk from "redux-thunk";
import Immutable from "immutable";
import configureMockStore from 'redux-mock-store';
import ClientHttpError from "./ClientHttpError";

export const findByTestAttribute = (component,attr) => component.find(`[data-test="${attr}"]`);

export const testStore = state => configureMockStore()(state);

export const clearKeys = (tree, arr) => {
  if (arr && arr.indexOf(tree) > -1) {
    return;
  }
  if (tree && typeof tree === 'object') {
    if (tree.inputId) {
      return;
    }
    arr.push(tree)
    if (tree.key) {
      try {
        delete tree.key;
      } catch (e) {}
    }
    if (Array.isArray(tree)) {
      tree.forEach(obj => clearKeys(obj, arr));
    } else {
      Object.keys(tree).forEach(key => clearKeys(tree[key], arr));
    }
  }
}

export function setupFetchStub(data) {
  return function fetchStub(_url) {
    return new Promise((resolve) => {
      resolve({
        json: () =>
          Promise.resolve({
            data,
          }),
      })
    })
  }
}

export function setupFetchThrowStub() {
  const error = new ClientHttpError(500, "test-error"); // Directly throwing new ClientHttpError fails in Jest
  throw error;
}
