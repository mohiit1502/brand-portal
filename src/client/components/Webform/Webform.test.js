import React, {useRef} from "react";
import Webform from "./Webform";
import toJson from "enzyme-to-json";
import {configure, mount} from "enzyme";
import {Provider} from "react-redux";
import MockNextContext from "../../../../test/client/utility/MockNextContext";
import {clearKeys, testStore} from "../../utility/TestingUtils";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
import Http from "../../utility/Http";

configure({adapter: new Adapter()});

let props;
let store;

const setUp = (props) => {
  store = testStore({modal: {}});
  return renderer.create(props ? <Provider store={store}><MockNextContext pathname="/help"><Webform {...props} /></MockNextContext></Provider> : <Provider store={store}><MockNextContext><Webform /></MockNextContext></Provider>);
};

const setUpMounted = () => {
  const mockStore = {
    modal: {}
  };
  store = testStore(mockStore);
  return mount(<Provider store={store}><Webform/></Provider>);
};

describe("Webform", () => {
  let wrapper;
  beforeEach(() => {
    jest.restoreAllMocks();
    jest.resetAllMocks();
  });
  it("renders without error", () => {
    props = {
      configuration: {},
      dispatchMetadata: ()=>{},
      dispatchWebformState: ()=>{},
      showNotification: ()=>{},
      toggleModal: ()=>{},
      webformFieldsConfiguration: {}
    };
    wrapper = setUp(props);

    const tree = wrapper.toJSON();
    clearKeys(tree, []);
    // expect(tree).toMatchSnapshot();
  });
  it("renders the Webform without error", () => {
    jest.spyOn(Http, "get").mockImplementation(() => Promise.resolve({body: {}}));
    wrapper = setUpMounted();
    toJson(wrapper);
    // expect(tree).toMatchSnapshot();
  });
});
