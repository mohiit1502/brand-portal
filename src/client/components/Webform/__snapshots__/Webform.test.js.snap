// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webform renders without error 1`] = `
<div
  className="c-Webform mt-4rem d-flex justify-content-center "
>
  <div
    className="col-lg-8 col-md-6 col-6 pl-3 pr-3"
  >
    <div />
    
    <div
      className="web-form mb-4 mr-3"
    >
      <div
        className="form-row font-weight-bold"
      >
        <div
          className=" col"
        >
          <div
            className="c-HeaderFormComponent row form-prompt font-weight-bold"
          >
            <div
              className="col"
            >
              <p>
                Filing Claim As
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-row mb-3"
      >
        <div
          className=" col-6"
        >
          <div
            className="form-group custom-input-form-group custom-select-form-group dropdown  mb-3  field-select-arrow"
          >
            <img
              alt="i"
              className="positioned-top-right"
              data-html="true"
              data-placement="right"
              data-toggle="tooltip"
              src="question.svg"
              title="<div class=\\"py-2\\" data-reactroot=\\"\\"><b class=\\"position-absolute text-white tooltip-close-button\\">x</b><p class=\\"mt-2 pl-2 text-left font-size-12\\"><span><br/><b>Rights Owner<!-- -->:</b> <!-- -->Brand owners with registered trademarks.<br/></span><span><br/><b>Third Party representative<!-- -->:</b> <!-- -->Authorized third-party brand protection agencies/ Legal representatives.<br/></span><span><br/><b>Customer<!-- -->:</b> <!-- -->Walmart customer.<br/></span></p></div>"
            />
            <input
              autoComplete="off"
              className="form-control form-control-userType custom-input-element"
              data-toggle="dropdown"
              disabled={false}
              id="webform-userType-custom-input"
              onChange={[Function]}
              required={false}
              type="select"
              value=""
            />
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="webform-userType-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text"
              >
                 
                User Type
                 
              </span>
            </label>
            <div
              className="dropdown-menu"
            >
              <a
                className="dropdown-item"
                onClick={[Function]}
              >
                Customer
              </a>
              <a
                className="dropdown-item"
                onClick={[Function]}
              >
                Rights Owner
              </a>
              <a
                className="dropdown-item"
                onClick={[Function]}
              >
                Third party
              </a>
            </div>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": "0.7rem",
                }
              }
            >
              
            </small>
          </div>
        </div>
      </div>
      <div
        className="form-row font-weight-bold"
      >
        <div
          className=" col"
        >
          <div
            className="c-HeaderFormComponent row form-prompt font-weight-bold"
          >
            <div
              className="col"
            >
              <p>
                Contact Information
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className=" col-6"
        >
          <div
            className="form-group custom-input-form-group form-group-text mb-3
        "
            style={
              Object {
                "position": "static",
              }
            }
          >
            <input
              className="form-control form-control-firstName custom-input-element"
              disabled={false}
              id="webform-firstName-custom-input"
              onBlur={[Function]}
              onChange={[Function]}
              onInvalid={[Function]}
              required={false}
              type="text"
              value=""
            />
            
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="webform-firstName-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text required"
              >
                 
                First Name
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              
            </small>
          </div>
        </div>
        <div
          className=" col-6"
        >
          <div
            className="form-group custom-input-form-group form-group-text mb-3
        "
            style={
              Object {
                "position": "static",
              }
            }
          >
            <input
              className="form-control form-control-lastName custom-input-element"
              disabled={false}
              id="webform-lastName-custom-input"
              onBlur={[Function]}
              onChange={[Function]}
              onInvalid={[Function]}
              required={false}
              type="text"
              value=""
            />
            
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="webform-lastName-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text required"
              >
                 
                Last Name
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              
            </small>
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className=" col-6"
        >
          <div
            className="form-group custom-input-form-group form-group-text mb-3
        "
            style={
              Object {
                "position": "static",
              }
            }
          >
            <input
              className="form-control form-control-address_1 custom-input-element"
              disabled={false}
              id="webform-address_1-custom-input"
              onBlur={[Function]}
              onChange={[Function]}
              onInvalid={[Function]}
              required={false}
              type="text"
              value=""
            />
            
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="webform-address_1-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text required"
              >
                 
                Address 1
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              
            </small>
          </div>
        </div>
        <div
          className=" col-6"
        >
          <div
            className="form-group custom-input-form-group form-group-text mb-0
        "
            style={
              Object {
                "position": "static",
              }
            }
          >
            <input
              className="form-control form-control-address_2 custom-input-element"
              disabled={false}
              id="webform-address_2-custom-input"
              onBlur={[Function]}
              onChange={[Function]}
              onInvalid={[Function]}
              required={false}
              type="text"
              value=""
            />
            
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="webform-address_2-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text"
              >
                 
                Address 2
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              Optional
            </small>
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className=" col-6"
        >
          <div
            className="form-group custom-input-form-group form-group-text mb-3
        "
            style={
              Object {
                "position": "static",
              }
            }
          >
            <input
              className="form-control form-control-city custom-input-element"
              disabled={false}
              id="webform-city-custom-input"
              onBlur={[Function]}
              onChange={[Function]}
              onInvalid={[Function]}
              required={false}
              type="text"
              value=""
            />
            
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="webform-city-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text required"
              >
                 
                City
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              
            </small>
          </div>
        </div>
        <div
          className=" col-6"
        >
          <div
            className="form-group custom-input-form-group form-group-text mb-3
        "
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <input
              className="form-control form-control-country custom-input-element"
              disabled={false}
              id="webform-country-custom-input"
              onBlur={[Function]}
              onChange={[Function]}
              onInvalid={[Function]}
              required={false}
              type="text"
              value="USA"
            />
            <label
              className="custom-input-label"
              htmlFor="webform-country-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text required"
              >
                 
                Country
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              
            </small>
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className="col-6 col-6"
        >
          <div
            className="form-group custom-input-form-group form-group-text mb-3
        "
            style={
              Object {
                "position": "static",
              }
            }
          >
            <input
              className="form-control form-control-state custom-input-element"
              disabled={false}
              id="webform-state-custom-input"
              onBlur={[Function]}
              onChange={[Function]}
              onInvalid={[Function]}
              required={false}
              type="text"
              value=""
            />
            
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="webform-state-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text required"
              >
                 
                State
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              
            </small>
          </div>
        </div>
        <div
          className="col-6 col-6"
        >
          <div
            className="form-group custom-input-form-group form-group-text mb-3
        "
            style={
              Object {
                "position": "static",
              }
            }
          >
            <input
              className="form-control form-control-zip custom-input-element"
              disabled={false}
              id="webform-zip-custom-input"
              maxLength={10}
              onBlur={[Function]}
              onChange={[Function]}
              onInvalid={[Function]}
              required={false}
              type="text"
              value=""
            />
            
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="webform-zip-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text required"
              >
                 
                ZIP
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              
            </small>
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className=" col-6"
        >
          <div
            className="form-group custom-input-form-group form-group-text mb-3
        "
            style={
              Object {
                "position": "static",
              }
            }
          >
            <input
              className="form-control form-control-phone custom-input-element"
              disabled={false}
              id="webform-phone-custom-input"
              maxLength={15}
              onChange={[Function]}
              onInvalid={[Function]}
              required={false}
              type="text"
              value=""
            />
            
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="webform-phone-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text required"
              >
                 
                Phone Number
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              
            </small>
          </div>
        </div>
        <div
          className=" col-6"
        >
          <div
            className="form-group custom-input-form-group form-group-text mb-3
        "
            style={
              Object {
                "position": "static",
              }
            }
          >
            <input
              className="form-control form-control-emailId custom-input-element"
              disabled={false}
              id="webform-emailId-custom-input"
              onChange={[Function]}
              onInvalid={[Function]}
              required={false}
              type="text"
              value=""
            />
            
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="webform-emailId-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text required"
              >
                 
                Email Address
                 
              </span>
            </label>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": undefined,
                }
              }
            >
              
            </small>
          </div>
        </div>
      </div>
      <div
        className="form-row font-weight-bold"
      >
        <div
          className=" col-6"
        >
          <div
            className="c-HeaderFormComponent row form-prompt font-weight-bold"
          >
            <div
              className="col"
            >
              <p>
                Type of infringement
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-row mb-3"
      >
        <div
          className=" col-6"
        >
          <div
            className="form-group custom-input-form-group custom-select-form-group dropdown  mb-3  field-select-arrow"
          >
            <input
              autoComplete="off"
              className="form-control form-control-claimType custom-input-element"
              data-toggle="dropdown"
              disabled={false}
              id="webform-claimType-custom-input"
              onChange={[Function]}
              required={true}
              type="select"
              value=""
            />
            <label
              className="custom-input-label custom-input-label-placeholder"
              htmlFor="webform-claimType-custom-input"
            >
              <div
                className="label-upper-bg position-absolute w-100 h-50 d-block"
              />
              <div
                className="label-lower-bg position-absolute w-100 h-50 d-block"
              />
              <span
                className="label-text"
              >
                 
                Claim type
                 
              </span>
            </label>
            <div
              className="dropdown-menu"
            >
              <a
                className="dropdown-item"
                onClick={[Function]}
              >
                Trademark
              </a>
              <a
                className="dropdown-item"
                onClick={[Function]}
              >
                Patent
              </a>
              <a
                className="dropdown-item"
                onClick={[Function]}
              >
                Counterfeit
              </a>
              <a
                className="dropdown-item"
                onClick={[Function]}
              >
                Copyright
              </a>
            </div>
            <small
              className="form-text custom-input-help-text text-muted"
              style={
                Object {
                  "paddingLeft": "0.7rem",
                }
              }
            >
              
            </small>
          </div>
        </div>
      </div>
      <div
        className="form-row font-weight-bold mt-2"
      />
      <div
        className="form-row"
      />
      <div
        className="form-row"
      />
      <div
        className="form-row font-weight-bold"
      />
      <div
        className="form-row"
      />
      <div
        className="form-row"
      />
      <div
        className="form-row"
      />
      <div
        className="form-row font-weight-bold mt-2"
      />
      <div
        className="form-row"
      />
      <div
        className="form-row font-weight-bold"
      />
    </div>
  </div>
</div>
`;
